collect_app/src/main/java/org/odk/collect/android/database/forms/DatabaseFormsRepository.java

```diff
@@ -38,10 +38,10 @@
 public class DatabaseFormsRepository implements FormsRepository {
 
     private final StoragePathProvider storagePathProvider;
-    private final Clock clock;
+    private final Supplier<Long> clock;
     private final FormsDatabaseProvider formsDatabaseProvider;
 
-    public DatabaseFormsRepository(Clock clock, StoragePathProvider storagePathProvider, FormsDatabaseProvider formsDatabaseProvider) {
+    public DatabaseFormsRepository(Supplier<Long> clock, StoragePathProvider storagePathProvider, FormsDatabaseProvider formsDatabaseProvider) {
```

Comment:
This change will work assuming that we create a repository object every time we want to use it. OK maybe now this is the case but what if at some point someone reuses an already created object? Isn't it a bit dangerous?
What's the advantage of such a change?

Reply:
The advantage for me is using a standard library abstraction rather than a custom one - we can probably get rid of `Clock` tbh. What is the worry around the object being reused?

Reply:
The worry is that if you pass long value when you create an object it's a constant and if you reuse the object later that value is not you current timestamp anymore so wrong timestamps might be saved to the database?

Reply:
Ahh my mistake that's not the way it works probably no you are wondering what I'm talking about sorry it won't be outdated.

Reply:
Ah I see what you were worried about! Right yeah as you say, we still get a different value each time, so we're ok. Definitely a good one to look out for though.